src/charm.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""GLAuth Operator Charm."""
+
+import logging
+
+from charms.operator_libs_linux.v1 import snap
+from glauth import Glauth
```

Comment:
If you import `Glauth` like this, you are going to need the *__init__.py* file that you have in the _src/_ directory. Python uses the *__init__.py* file to determine if a directory is an importable namespace when not using relative imports. If you change the import to the following, you should not need the *__init__.py* file in _src/_:

```suggestion
from .glauth import Glauth
```

_Note that you will have to resort the imports with ruff since relative imports are supposed to go at the end of the global import block._

Reply:
Right, I added the init.py to avoid relative imports as I think most style guides recommend against using them (off the top of my head I know Google style guide does). I took a look at the juju sdk charm style guide and it doesn't mention anything but I did notice other Canonical charms don't do this so I want to avoid it for consistency sake.

Removed init.py and everything seems to be working fine.